packages/hash/api/src/db/postgres/link/getEntityOutgoingLinks.ts

```diff
@@ -36,3 +56,59 @@ export const getEntityOutgoingLinks = async (
 
   return rows.map(mapDBLinkRowToDBLink);
 };
+
+const outgoingLinkAggregationQuery = sql`select array_agg(destination_entity_id) as outgoing_entity_ids from links
+where a.entity_id = source_entity_id
+group by a.entity_id`;
+
+/**
+ * Get the latest version of all entities of a given type and their outgoing link ids with a matching entity type.
+ * @param params.entityTypeId the entity type id to return entities of
+ * @param params.entityTypeVersionId optionally limit to entities of a specific version of a type
+ * @param params.accountId the account to retrieve entities from
+ */
+export const getEntitiesByTypeWithOutgoingEntityIds = async (
+  conn: Connection,
+  params: {
+    entityTypeId: string;
+    entityTypeVersionId?: string;
+    accountId: string;
+  },
+): Promise<EntityWithOutgoingEntityIds[]> => {
+  const rows = await conn.any<EntityWithOutgoingEntityIdsPGRow>(sql`
+    with all_matches as (
+      ${selectEntitiesByType(params)}
+    )
+    , distinct_entitites as (select distinct on (entity_id) * from all_matches order by entity_id, updated_at desc)
+    select * from distinct_entitites as a
+    left join lateral (${outgoingLinkAggregationQuery}) as l on true
+    
+  `);
+  return rows.map(mapEntityWithOutgoingEntityIdsPGRowToEntity);
+};
+
+/**
+ * Get the latest version of a given entity and all outgoing link ids with a matching entity type.
+ * @param params.accountId the account to retrieve entities from
+ * @param params.entityId entityId of source entity.
+ */
+export const getEntityWithOutgoingEntityIds = async (
```

Comment:
Is there any potential for consolidating logic/query fragments between this and `getEntityOutgoingLinks`?